tarofreighter/chain_porter.go

```diff
@@ -1072,6 +1071,12 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 			return nil, err
 		}
 
+		// With the transaction broadcast, we'll deliver a
+		// notification via the transaction broadcast response channel.
+		if txBroadcastRespChan != nil {
+			currentPkg.deliverResponse(txBroadcastRespChan)
```

Comment:
I think if we embed the request within the `sendPackage`, then we won't need to modify any of the function signatures here.

Reply:
Made this change in the latest commits.